server/belga/io/feed_parsers/belga_afp_newsml_1_2.py

```diff
@@ -51,8 +51,17 @@ def parser_newsitem(self, item, newsitem_el):
             })
         # add content for headline when it is empty
         if item.get('urgency') in ('1', '2') and not item.get('headline'):
-            first_line = item.get('body_html', '').strip().split('\n')[0]
-            first_line = etree.fromstring(first_line).text
+            first_line = ''
+            # remove trailing whitespace in p or h2 tag and empty tag
+            body = re.sub(r'(>)(\s*)(.*?)(\s*)(<)', r'\1\3\5', item.get('body_html', ''))
```

Comment:
it's kind of hard to understand all those regexes, what about something like:
```python
from superdesk.text_utils import get_text

for line in get_text(body).split('\n'):
    if line.strip():
        item['headline'] = 'URGENT: {}'.format(line.strip())
        break
```

Reply:
There is a small problem, get_text will combine all p tag text as a single paragraph, for example, from the ticket attachment files
`
<body.content><p>Un homme poignarde des passants à Villejuif, la police le "neutralise" (préfecture de police)</p><p>sm/jt/bma</p><p/></body.content>
`

so get_text will return
`Un homme poignarde des passants à Villejuif, la police le "neutralise" (préfecture de police)sm/jt/bma
`

but the content includes two paragraphs, so we think the headline should be
`Un homme poignarde des passants à Villejuif, la police le "neutralise" (préfecture de police)
`

Reply:
you can use this [lf_on_block](https://github.com/superdesk/superdesk-core/blob/develop/superdesk/text_utils.py#L45)

Reply:
Sorry we missed that, with it  `get_text` is working as we expected :+1: